mm2src/coins/lp_coins.rs

```diff
@@ -686,6 +689,66 @@ pub async fn lp_coinfindᵃ(ctx: &MmArc, ticker: &str) -> Result<Option<MmCoinEn
     Ok(coins.get(ticker).cloned())
 }
 
+#[derive(Deserialize)]
+struct ConvertAddressReq {
+    coin: String,
+    from: String,
+    /// format to that the input address should be converted
+    to_address_format: Json,
+}
+
+pub async fn convert_address(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let req: ConvertAddressReq = try_s!(json::from_value(req));
+    let coin = match lp_coinfindᵃ(&ctx, &req.coin).await {
+        Ok(Some(t)) => t,
+        Ok(None) => return ERR!("No such coin: {}", req.coin),
+        Err(err) => return ERR!("!lp_coinfind({}): {}", req.coin, err),
+    };
+    let result = json!({
+        "result": {
+            "address": try_s!(coin.convert_to_address(&req.from, req.to_address_format)),
+        },
+    });
+    let body = try_s!(json::to_vec(&result));
+    Ok(try_s!(Response::builder().body(body)))
+}
+
+#[derive(Deserialize)]
+struct ValidateAddressReq {
+    coin: String,
+    address: String,
+}
+
+#[derive(Serialize)]
+struct ValidateAddressResult {
+    is_valid: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    reason: Option<String>,
+}
+
+pub async fn validate_address(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let req: ValidateAddressReq = try_s!(json::from_value(req));
+    let coin = match lp_coinfindᵃ(&ctx, &req.coin).await {
+        Ok(Some(t)) => t,
+        Ok(None) => return ERR!("No such coin: {}", req.coin),
+        Err(err) => return ERR!("!lp_coinfind({}): {}", req.coin, err),
+    };
+
+    let result = match coin {
```

Comment:
I believe you wanted to avoid code duplication in new method trait method (e.g. `validate_address`). And it's impossible to add `address_from_str` to the trait because `address_from_str` return different types per coin.
Maybe we can add a macro for `validate_address` trait method implementation?